business/istio_status.go

```diff
@@ -223,3 +210,45 @@ func GetDeploymentStatus(d apps_v1.Deployment) string {
 	}
 	return status
 }
+
+func (iss *IstioStatusService) getAddonComponentStatus() IstioComponentStatus {
+	extServices := config.Get().ExternalServices
+	ics := IstioComponentStatus{}
+
+	ics.merge(getAddonStatus("prometheus", true, extServices.Prometheus.URL, &extServices.Prometheus.Auth, true))
+	ics.merge(getAddonStatus("grafana", extServices.Grafana.Enabled, extServices.Grafana.InClusterURL, &extServices.Grafana.Auth, extServices.Grafana.IsCoreComponent))
+	ics.merge(getAddonStatus("jaeger", extServices.Tracing.Enabled, extServices.Tracing.InClusterURL, &extServices.Tracing.Auth, extServices.Tracing.IsCoreComponent))
+	ics.merge(getAddonStatus("custom dashboards", extServices.CustomDashboards.Enabled, extServices.CustomDashboards.Prometheus.URL, &extServices.CustomDashboards.Prometheus.Auth, extServices.CustomDashboards.IsCoreComponent))
```

Comment:
There's a subtlety here, because custom dashboards can use the main prometheus if the custom one isn't configured. So it must be something like this:

```suggestion
        customProm := extServices.CustomDashboards.Prometheus
        if customProm.URL == "" {
            customProm = extServices.Prometheus
        }
	ics.merge(getAddonStatus("custom dashboards", extServices.CustomDashboards.Enabled, customProm.URL, &customProm.Auth, extServices.CustomDashboards.IsCoreComponent))
```

Reply:
Good! And I also realized that I didn't paralelized the calls. I've add it to.